### PR TITLE
Add `Lint/PercentStringArray` and `Lint/PercentSymbolArray` cops

### PR DESCRIPTION
**Pull request by owst**
Adds a new cop `Lint/PercentString` cop:

      # This cop checks for colons and commas in %i, e.g.
      #
      #   `%i(:foo, :bar)`
      #
      # and quotes and commas in %w, e.g.
      #
      #   `%w('foo', "bar")`
      #
      # in both cases, it is more likely that the additional characters are
      # unintended (for example, mistranslating an array of literals to percent
      # string notation) rather than meant to be part of the resulting
      # strings/symbols.

I recently made this mistake in my work code, and it was only due to a keen-eyed reviewer (@Qlexander - thanks 👍) that I avoided introducing a bug, so I think Rubocop can help us out here.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it)
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

File: lib/rubocop/cop/style/global_vars.rb [hunk 1]
```diff
@@ -15,6 +15,7 @@ class GlobalVars < Cop
 
         # predefined global variables their English aliases
         # http://www.zenspider.com/Languages/Ruby/QuickRef.html
+        # rubocop:disable Lint/PercentString
```
Comment (owst) [inline lib/rubocop/cop/style/global_vars.rb]: Because of `$,` on line 26

**Comment by joehorsnell** [comment]
Looks good @owst - only minor comments would be to a) potentially call it `Lint/PercentArrayLiteral` since they are literal array rather than string syntax and b) maybe have two linters, one for the string array and the other for the symbol array.

**Comment by bbatsov** [comment]
@joehorsnell Great remarks. Probably I'd go with different cops.

File: lib/rubocop/cop/style/global_vars.rb (hunk 1 above)
Comment (bbatsov) [inline lib/rubocop/cop/style/global_vars.rb]: I wonder if we can account account for cases like these.

**Comment by owst** [comment]
About naming - the literals ruby doc refers to them as [Percent Strings](http://ruby-doc.org/core-2.3.0/doc/syntax/literals_rdoc.html#label-Percent+Strings)

**Comment by owst** [comment]
@bbatsov separate cops - done!

**Comment by bbatsov** [comment]
Has to be rebased.

**Comment by owst** [comment]
@bbatsov - done!

**Pull request merged**